lite2/client.go

```diff
@@ -649,15 +650,34 @@ func (c *Client) cleanup(stopHeight int64) error {
 		return errors.Wrap(err, "can't get last trusted height")
 	}
 
-	// 3) Remove all headers and validator sets.
-	if stopHeight < oldestHeight {
-		stopHeight = oldestHeight
+	// 3) Ensure the start height and stop height are within the range of headers stored.
+	if stopHeight == 0 || stopHeight > latestHeight {
+		stopHeight = latestHeight
+	}
+	if startHeight < oldestHeight {
+		startHeight = oldestHeight
 	}
-	for height := stopHeight; height <= latestHeight; height++ {
-		err = c.trustedStore.DeleteSignedHeaderAndValidatorSet(height)
+	if startHeight > stopHeight {
+		return errors.Errorf("startHeight of %d must be less than stopHeight %d", startHeight, stopHeight)
```

Comment:
Is this a developer error?

Reply:
yes, the function could work without it - perhaps log it as an error or we can also remove it